fe1-web/model/objects/WalletCryptographyHandler.ts

```diff
@@ -0,0 +1,76 @@
+import { IndexedDBStore, WalletCryptoKey } from 'store/stores/IndexedDBStore';
+import STRINGS from 'res/strings';
+
+/**
+ * This class has the job of handling the cryptography functions of the wallet.
+ * It interacts with the IndexedDB database in order to store and retrieve the
+ * secret key. It also encrypts and decrypts the tokens with the retrieved key.
+ */
+export class WalletCryptographyHandler {
+  /* encryption/decryption algorithm (RSA) */
+  private readonly algorithm = {
+    name: 'RSA-OAEP',
+    modulusLength: 4096,
+    publicExponent: new Uint8Array([1, 0, 1]),
+    hash: 'SHA-256',
+  };
+
+  private readonly keyUsages: KeyUsage[] = ['encrypt', 'decrypt'];
+
+  /* the encryption/decryption key storage */
+  private keyDatabase: IndexedDBStore = new IndexedDBStore();
+
+  /**
+   * this method adds an RSA encryption/decryption key,
+   * interacts with IndexedDB browser database.
+   */
+  public async initWalletStorage(): Promise <void> {
+    const key: WalletCryptoKey = await this.generateRSAKey();
+    await this.keyDatabase.putKey(key);
+  }
+
+  /**
+   * encrypts the given ed25519 token with the RSA key stored in the indexedDB database
+   * @param token ed25519 key toUint8Array()
+   */
+  public async encrypt(token: Uint8Array): Promise<ArrayBuffer> {
+    const key = await this.getKeyFromDatabase(STRINGS.walletPublicKey);
+    const cypheredToken = await window.crypto.subtle.encrypt(this.algorithm, key, token);
+    return cypheredToken;
+  }
+
+  /**
+   * decrypts the encrypted ed25519 token with the RSA key stored in the indexedDB database
+   * @param encryptedToken ed25519 encrypted token (ArrayBuffer)
+   */
+  public async decrypt(encryptedToken: ArrayBuffer): Promise<ArrayBuffer> {
+    const key = await this.getKeyFromDatabase(STRINGS.walletPrivateKey);
+    const plaintextToken = await window.crypto.subtle.decrypt(this.algorithm, key, encryptedToken);
+    return plaintextToken;
```

Comment:
Careful here @RaulinN , you might be a tiny bit mistaken :) 
The `await/async` is a way of having more readable promise chaining. We haven't used it so far, but it's a perfectly valid approach :) 

The whole function is basically the equivalent of doing:

```
  public async decrypt(encryptedToken: ArrayBuffer): Promise<ArrayBuffer> {
    return this.getKeyFromDatabase(STRINGS.walletPrivateKey)
      .then((key) => window.crypto.subtle.decrypt(this.algorithm, key, encryptedToken));
```

Your version has a bug, as you'd return the keyPromise and not the decryption results.

Reply:
Yes indeed! Misread the documentation about await/async ^^' Sorry for that!

For the piece of code I wrote, I didn't expect it to follow javascript's syntax since I was very busy with Kotlin's futures lately ^^ (that's also why I put the disclaimer "approximate syntax")